### ISSUE TITLE
[REGRESSION]: Component tests: input element is not mounted

### ISSUE THREAD

**Issue by vitalets** [microsoft/playwright#28566]
### System info
- Playwright Version: v1.40
- Operating System: macOS 13.4.1
- Browser: All

### Source code

**This works:**
```tsx
import React from 'react';
import { test, expect } from '@playwright/experimental-ct-react';

test('should work', async ({ mount, page }) => {
  const component = await mount(<textarea/>);
  console.log(await page.content())
  await expect(component).toBeVisible();
});
```
<details>
<summary>Output:</summary>

```
$ npm run test-ct

> test-ct
> playwright test -c playwright-ct.config.ts

Running 1 test using 1 worker
[chromium] › App.spec.tsx:4:5 › should work
<!DOCTYPE html><html lang="en"><head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Testing Page</title>
    <script type="module" crossorigin="" src="/assets/index-948b91f8.js"></script>
  </head>
  <body>
    <div id="root"><textarea></textarea></div>
</body></html>
  1 passed (648ms)
```
</details>

**This does NOT work:**
```tsx
import React from 'react';
import { test, expect } from '@playwright/experimental-ct-react';

test('should work', async ({ mount, page }) => {
  const component = await mount(<input/>); // <- notice input instead of textarea
  console.log(await page.content())
  await expect(component).toBeVisible();
});
```
<details>
<summary>Output:</summary>

```
npm run test-ct

> test-ct
> playwright test -c playwright-ct.config.ts

Running 1 test using 1 worker
[chromium] › App.spec.tsx:4:5 › should work
<!DOCTYPE html><html lang="en"><head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Testing Page</title>
    <script type="module" crossorigin="" src="/assets/index-948b91f8.js"></script>
  </head>
  <body>
    <div id="root"></div>
</body></html>
  1) [chromium] › App.spec.tsx:4:5 › should work ───────────────────────────────────────────────────

    Test timeout of 1000ms exceeded.
```
</details>

**Steps**
- Setup component testing for React [per docs](https://playwright.dev/docs/test-components#how-to-get-started)
- Use tests from above examples

**Expected**
`<input>` element is mounted.

**Actual**
`<input>` element is NOT mounted.

NOTE: it is reproduced only on Playwright 1.40. In Playwright 1.39 and 1.41 (next) it works fine.

**Comment by mxschmitt** [comment]
cc @sand4rt

**Comment by pavelfeldman** [comment]
Closing as fixed on 1.41

**Comment by lukaskoeller** [comment]
Still not fixed with `1.40.1`. Is duplicated by https://github.com/microsoft/playwright/issues/28666 but with updated playwright version and reproduction repo
